backend/main.py

```
@@ -1388,6 +1390,38 @@ async def connect_codebase(request: CodebaseConnectRequest):
+        logger.info(f"Total monthly LLM cost across all projects: ${total_cost:.6f} ({len(llm_records)} calls)")
+        
+        return {
+            "total_cost": round(total_cost, 6),
+            "call_count": len(llm_records),
+            "year": current_year,
+            "month": current_month
+        }
+        
+    except Exception as e:
+        logger.error(f"Error getting monthly LLM cost: {e}")
+        raise HTTPException(status_code=500, detail=f"Failed to get monthly LLM cost: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)